### PR TITLE
(#12) Implement the zenboard sharing feature 

### LINKED ISSUE #12: [Feat]: Implement sharing ability for users on your zenboard & html preview.
Checkout the share page on the figma
Write the documentation

### LINKED ISSUE #18: [Fix]: Rework sign-in with github to use passport-github

### PR DESCRIPTION
**Pull request by Alfred-Emmanuel**
feat: closes #12 and fix: closes #18

### CODE REVIEW

File: src/zenboard/models/permissions.ts [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+  },
+  zenboardId: {
+    type: Schema.Types.ObjectId,
+    ref: "Zenboard",
+    required: true,
+  },
+  accessLevel: {
+    type: String,
+    enum: ["owner", "editor", "viewer"],
+    default: "viewer",
```
Comment (tdadadavid) [inline src/zenboard/models/permissions.ts:18]: create a typescript enum for this you can call it `AccessLevels`

File: src/zenboard/routes/index.ts [hunk 2]
```diff
@@ -1,2 +1,93 @@
...
+  //   const ownerId = req.user?.id;
+
+  if (!title) {
+    return res.status(400).json({ message: "Title and ownerId are required" });
+  }
+
+  try {
+    const zenboard = await Zenboard.create({
+      title,
+      //   ownerId,
```
Comment (tdadadavid) [inline src/zenboard/routes/index.ts]: Why are not attaching the board to the particular user that created it.

File: src/zenboard/models/index.ts [hunk 3]
```diff
@@ -1 +1,30 @@
-// each model will be separated in files and exported here
+import { model, Schema } from "mongoose";
+
+const zenboardSchema = new Schema(
+  {
+    // ownerId: {
```
Comment (tdadadavid) [inline src/zenboard/models/index.ts]: Why did you leave this commented out.

File: src/zenboard/services/permissions.ts [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+  zenboardId: Types.ObjectId,
+  accessLevel: "owner" | "editor" | "viewer" = "viewer"
+) => {
+  const permission = new Permissions({ userId, zenboardId, accessLevel });
+  return await permission.save();
+};
+
+export const updatePermission = async (
+  permissionId: Types.ObjectId,
+  accessLevel: "owner" | "editor" | "viewer"
```
Comment (tdadadavid) [inline src/zenboard/services/permissions.ts:15]: Change this to a type or enum as I said earlier.

**Review by tdadadavid — CHANGES_REQUESTED** [review]

**Comment by Alfred-Emmanuel** [comment]
@tdadadavid all done

File: src/auth/helpers/token.service.ts [hunk 5]
```diff
@@ -90,4 +223,13 @@ export class TokenService {
       jwtid: crypto.randomUUID(),
     });
   }
+
+  // Helper methods to check if token is from Google or GitHub
+  private isGoogleToken(token: string): boolean {
```
Comment (tdadadavid) [inline src/auth/helpers/token.service.ts:229]: What is the thought process behind this?

**Review by tdadadavid — CHANGES_REQUESTED** [review]

**Review by tdadadavid — APPROVED** [review]

**Pull request merged**